server/src/main/java/org/opensearch/repositories/Repository.java

```diff
@@ -250,6 +251,42 @@ void snapshotShard(
         ActionListener<String> listener
     );
 
+    /**
+     * Adds a reference of remote store data for a index commit point.
+     * <p>
+     * The index commit point can be obtained by using {@link org.opensearch.index.engine.Engine#acquireLastIndexCommit} method.
+     * Repository implementations shouldn't release the snapshot index commit point. It is done by the method caller.
+     * <p>
+     * As snapshot process progresses, implementation of this method should update {@link IndexShardSnapshotStatus} object and check
+     * {@link IndexShardSnapshotStatus#isAborted()} to see if the snapshot process should be aborted.
+     * @param store                    store to be snapshotted
+     * @param mapperService            the shards mapper service
+     * @param snapshotId               snapshot id
+     * @param indexId                  id for the index being snapshotted
+     * @param snapshotIndexCommit      commit point
+     * @param shardStateIdentifier     a unique identifier of the state of the shard that is stored with the shard's snapshot and used
+     *                                 to detect if the shard has changed between snapshots. If {@code null} is passed as the identifier
+     *                                 snapshotting will be done by inspecting the physical files referenced by {@code snapshotIndexCommit}
+     * @param snapshotStatus           snapshot status
+     * @param repositoryMetaVersion    version of the updated repository metadata to write
+     * @param userMetadata             user metadata of the snapshot found in {@link SnapshotsInProgress.Entry#userMetadata()}
+     * @param primaryTerm              current Primary Term
+     * @param listener                 listener invoked on completion
+     */
+    void snapshotRemoteStoreIndexShard(
```

Comment:
Rename generically to `checkpointShard`

Reply:
ack

Reply:
Also you might want to see how to fix existing breaking plugins like GCP/Azure/HDFS